modules/ding_permissions/ding_permissions.module

```diff
@@ -1000,12 +1001,10 @@ function ding_permissions_secure_permissions($role) {
  * Implements hook_preprocess_HOOK().
  */
 function ding_permissions_preprocess_html(&$variables) {
-  global $user;
```

Comment:
It makes sense to use permissions instead of roles but why is this check for `access administration menu AND NOT access administration pages` needed?

I would think the proper way to do this would be to avoid this preprocessing altogether and rely on the permissions already supported by the toolbar and administration modules: If the user has `access administration menu` then show the admin menu. If the user has `access toolbar` then show the toolbar.